Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/AutoMoveTo.java

```diff
@@ -1,36 +1,61 @@
 package org.usfirst.frc.team199.Robot2018.commands;
 
-import edu.wpi.first.wpilibj.command.Command;
+import org.usfirst.frc.team199.Robot2018.Robot;
+import org.usfirst.frc.team199.Robot2018.autonomous.AutoUtils;
+import org.usfirst.frc.team199.Robot2018.autonomous.PIDSourceAverage;
+
+//import org.usfirst.frc.team199.Robot2018.subsystems.Drivetrain;
+
+import edu.wpi.first.wpilibj.command.CommandGroup;
 
 /**
  *
  */
-public class AutoMoveTo extends Command {
-
-    public AutoMoveTo(String[] points) {
-        // Use requires() here to declare subsystem dependencies
-        // eg. requires(chassis);
-    }
-
-    // Called just before this Command runs the first time
-    protected void initialize() {
-    }
-
-    // Called repeatedly when this Command is scheduled to run
-    protected void execute() {
-    }
-
-    // Make this return true when this Command no longer needs to run execute()
-    protected boolean isFinished() {
-        return false;
-    }
-
-    // Called once after isFinished returns true
-    protected void end() {
-    }
-
-    // Called when another command which requires one or more of the same
-    // subsystems is scheduled to run
-    protected void interrupted() {
+public class AutoMoveTo extends CommandGroup {
+	
+    public AutoMoveTo(String[] args) {
+        //requires(Drivetrain);
+    	double rotation;
+    	double[] point = new double[2];
+    	String parentheseless;
+    	String[] pointparts;
+    	for (String arg : args) {
+    		if (AutoUtils.isDouble(arg)) {
+    			rotation = Double.valueOf(arg);
+    			addSequential(new PIDTurn(rotation - AutoUtils.getRot(), Robot.dt, Robot.dt.getGyro()));
+    			AutoUtils.setRot(rotation);
+    		} else if (AutoUtils.isPoint(arg)) {
+    			parentheseless = arg.substring(1, arg.length() - 1);
+    			pointparts = parentheseless.split(",");
+    			point[0] = Double.parseDouble(pointparts[0]);
+    			point[1] = Double.parseDouble(pointparts[1]);
+    			addSequential(new PIDTurn(Math.toDegrees(Math.atan((point[0] - AutoUtils.getX())/(point[1] - AutoUtils.getY())) - AutoUtils.getRot()), Robot.dt, Robot.dt.getGyro()));
+    			addSequential(new PIDMove(Math.sqrt(((point[0] - AutoUtils.getX()) * (point[0] - AutoUtils.getX()) + ((point[1] - AutoUtils.getY()) * (point[1] - AutoUtils.getY())))), Robot.dt, new PIDSourceAverage(null, null)));
+    			AutoUtils.setX(point[0]);
+    			AutoUtils.setY(point[1]);
+    			AutoUtils.setRot(Math.toDegrees(Math.atan((point[0] - AutoUtils.getX())/(point[1] - AutoUtils.getY()))));
+    		} else {	
+    			throw new IllegalArgumentException();
+    		}
+    	}
+    	
```

Comment:
instead of having all of the AutoUtils methods be static, can you pass an instance of it into the constructor of this command instead? We're trying to move away from statics because it's better for testing purposes, as well as dependency issues.

Reply:
I would argue that in this case, AutoUtils should be static as it doesn't represent an object and we don't need to reuse its parameters.

Reply:
However, if it really is making it hard to test, I can change it.